discovery/src/main/scala/akka/discovery/SimpleServiceDiscovery.scala

```diff
@@ -112,6 +112,54 @@ case object Lookup {
    * and protocol
    */
   def create(serviceName: String): Lookup = new Lookup(serviceName, None, None)
+
+  private val SrvQuery = """^_(.+?)\._(.+?)\.(.+?)$""".r
+
+  private val DomainName = "^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}$".r
+
+  /**
+   * Create a service Lookup from a string with format:
+   * _portName._protocol.serviceName.
+   * (as specified by https://www.ietf.org/rfc/rfc2782.txt)
+   *
+   * If the passed string conforms with this format, a SRV Lookup is returned.
+   *
+   * The string is parsed and dismembered to build a Lookup as following:
+   * Lookup(serviceName).withPortName(portName).withProtocol(protocol)
+   *
+   * If the string doesn't not conform with the SRV format, a simple A/AAAA Lookup is returned using the whole string as service name.
+   *
+   * @throws This method throws IllegalArgumentException if the service name is not a valid domain name.
```

Comment:
```suggestion
   * @throws IllegalArgumentException if the service name is not a valid domain name.
```

You have to put the exception class first.

Reply:
correct